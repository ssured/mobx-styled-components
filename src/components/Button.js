import styled from 'styled-components';
import { inject, observer } from 'mobx-react';

const Button = inject('AppStyle')(
  observer(
    styled.button`
      font-size: 1em;
      margin: 1em;
      padding: 0.25em 1em;
      border: 2px solid palevioletred;
      border-radius: 3px;
      background: ${({ AppStyle: { background } }) => background};
  `,
  ),
);

export default Button;
