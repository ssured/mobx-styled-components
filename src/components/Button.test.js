import React from 'react';
import { observable } from 'mobx';

import { shallow } from 'enzyme';

import Button from './Button';

describe('<Button />', () => {
  test('should render a button with a classname', () => {
    const AppStyle = observable({
      background: 'blue',
    });
    const wrapper = shallow(<Button AppStyle={AppStyle} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should render other class when style is provided', () => {
    const AppStyle = observable({
      background: 'blue',
    });
    const wrapper = shallow(<Button AppStyle={AppStyle} />);
    let html1, html2;
    expect(html1 = wrapper.html()).toMatchSnapshot();

    // trigger change in the global style
    AppStyle.background = 'red';
    expect(html2 = wrapper.html()).toMatchSnapshot();

    expect(html1).not.toEqual(html2);
  });
});
