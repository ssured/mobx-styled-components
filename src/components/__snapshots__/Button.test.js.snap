// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render a button with a classname 1`] = `"<button class=\\"jqttdJ\\"></button>"`;

exports[`<Button /> should render other class when style is provided 1`] = `"<button class=\\"jqttdJ\\"></button>"`;

exports[`<Button /> should render other class when style is provided 2`] = `"<button class=\\"jjqOFT\\"></button>"`;
